x/concentrated-liquidity/internal/swapstrategy/zero_for_one.go

```diff
@@ -35,33 +34,35 @@ func (s zeroForOneStrategy) GetSqrtTargetPrice(nextTickSqrtPrice sdk.Dec) sdk.De
 	return nextTickSqrtPrice
 }
 
-// ComputeSwapStep calculates the next sqrt price, the new amount remaining, the amount of the token other than remaining given current price, and total fee charge.
+// ComputeSwapStepLegacy is the old implementation of ComputeSwapStepOutGivenIn and ComputeSwapStepInGivenOut.
+// Calculates the next sqrt price, the new amount remaining, the amount of the token other than remaining given current price, and total fee charge.
 //
 // zeroForOneStrategy assumes moving to the left of the current square root price.
 // amountRemaining is the amount of token in when swapping out given in and token out when swapping in given out.
 // amountRemaining is token zero.
 // amountZero is token out when swapping in given out and token in when swapping out given in.
 // amountOne is token in when swapping in given out and token out when swapping out given in.
-// TODO: improve tests
-func (s zeroForOneStrategy) ComputeSwapStep(sqrtPriceCurrent, sqrtPriceNextTick, liquidity, amountRemaining sdk.Dec) (sdk.Dec, sdk.Dec, sdk.Dec, sdk.Dec) {
+//
+// To be removed in: https://github.com/osmosis-labs/osmosis/issues/4423
+func (s zeroForOneStrategy) ComputeSwapStepLegacy(sqrtPriceCurrent, sqrtPriceNextTick, liquidity, amountRemaining sdk.Dec) (sdk.Dec, sdk.Dec, sdk.Dec, sdk.Dec) {
```

Comment:
Note: similar idea to https://github.com/osmosis-labs/osmosis/pull/4500/files#r1125566291